content/docs/rfcs/37/README.md

```diff
@@ -0,0 +1,151 @@
+title: 37/WAKU2-NOISE-SESSIONS
+name: Session Management for Waku Noise
+status: raw
+tags: waku-core-protocol
+editor: Giuseppe <giuseppe@status.im>
+contributors: 
+---
+
+# Introduction
+
+In [35/WAKU2-NOISE](https://rfc.vac.dev/35/) we defined how Waku messages' payloads can be encrypted using key material derived from key agreements based on the [Noise Protocol Framework](http://www.noiseprotocol.org/noise.html). 
+
+Once two users complete a Noise handshake, 
+an encryption/decryption session - _or a Noise session_ - would be instantiated.
+
+This post provides an overview on how we can possibly implement and manage one or multiple Noise sessions in Waku.
+
+# Preliminaries
+
+We assume that two users, e.g. Alice and Bob, successfully completed a Noise handshake.
+
+Using [Noise terminology]((http://www.noiseprotocol.org/noise.html)), at the end of the handshake they will share:
+- two _Cipher States_ `CSOutbound` and `CSInbound`, to encrypt and decrypt outbound and inbound messages, respectively;
+- a handshake hash value `h`.
+
+As suggested in Noise specifications in regards to [Channel Binding](http://www.noiseprotocol.org/noise.html#channel-binding), 
+we can identify a Noise session with a `session-id` derived from the handshake hash value `h` shared on completion of a Noise handshake.
+
+More specifically, when Alice and Bob call [Split()](http://www.noiseprotocol.org/noise.html#the-symmetricstate-object) in order to derive the two final encryption and decryption Cipher States, 
+they further compute `session-id  = HKDF(h)` using the supported key derivation function `HKDF`.
+
+Such `session-id` will uniquely identify the Noise cryptographic session instantiated on completion of a Noise handshake,
+which would then consists of the tuple `(session-id, CSOutbound, CSInbound)`. 
+
+Once a Noise session is instantiated, 
+any further encrypted message between Alice and Bob within this session is exchanged on a `contentTopic` with name `/wakunoise/1/<ct-id>/proto`, 
+where `ct-id = Hash(Hash(session-id))`.
+
+
+# Session states
+
+A Noise session corresponding to a certain `session-id`:
+- is always **active** as long as it is not marked as **stale**. 
+For an active `session-id`, new messages are published on the content topic `/wakunoise/1/<ct-id>/proto`;  
+- is marked as **stale** if a [session termination message](https://rfc.vac.dev/spec/35/#session-termination-message) containing `Hash(session-id)` is published on the content topic `/wakunoise/1/<ct-id>/proto`. 
+Session information relative to stale sessions  MAY be deleted from users' device, unless required for later channel binding purposes.
+
+When a Noise session is marked as stale, it means that one party requested its termination while being online, 
+since publication of a hash pre-image for `ct-id` is required (i.e. `Hash(session-id)`).
+
+Currently, it is not possible to mark a Noise session as stale when `session-id` is lost or gets corrupted in users' devices.
```

Comment:
It is also not possible to communicate if the session-id or its H(H(session-id)) is lost

Reply:
Actually this sentence emphasize something different. The Noise session information, consisting of the `session-id` and the two Cipher states, is clearly required not only to agree on a place where parties communicate (the content topic) but to encrypt/decrypt messages in the first place: so persistence of `session-id` is as important as encryption keys when communicating.

Here, instead, we refer to the possibility to stale a session: if users lose session information (session-id + Cipher state) the session will remain active, at least until the other party takes some actions. But only `sessiond-id`is required (or, in principle `Hash(session-id)`) to publicly stale a session, rather than the full Noise session information.

Reply:
Thanks for the explanation, but my point was exactly the same, you have mentioned one but not the other i.e., the session id should be persisted
- To be able to communicate with the other party
- To be able to properly exit the communication with the other party
You have emphasized the second point, but not the first one.

Reply:
Not only the session-id, but the whole Noise session, that is session-id + Cipher states. I guess it is assumed that Noise session has to be persisted, since contains all the cryptographic information to either retrieve and process messages. This may be made explicit maybe earlier, but at this point, instead, the discussion is about staling sessions. Do you agree?

Reply:
Addressed in https://github.com/vacp2p/rfc/commit/f4c6b40ed3647120770a6330355c09e854d45e0c

Reply:
> This may be made explicit maybe earlier, but at this point, instead, the discussion is about staling sessions. Do you agree?

Sure, not right under this section, any place in the RFC that makes the most sense!